core/trino-main/src/main/java/io/trino/execution/RemoteTask.java

```diff
@@ -71,4 +72,13 @@
     PartitionedSplitsInfo getQueuedPartitionedSplitsInfo();
 
     int getUnacknowledgedPartitionedSplitCount();
+
+    /**
+     * Retrieves spooling output stats. Stats are available only for tasks that are in the {@link TaskState#FINISHED} state
+     * and have received the final task info (see {@link #addFinalTaskInfoListener(StateChangeListener)}.
+     * Stats can be retrieved only once and are discarded upon retrieval. Subsequent calls to the method after initial retrieval will fail.
```

Comment:
what is the reason for such semantics? It is unexpected for getter (`retrieve...` sounds like a getter) to modify the underlying state?
If we really need this kind of semantics maybe call the method `retriveAndDrop....`

Reply:
This is needed to avoid retaining the information for every task longer than necessary. In fault tolerant execution we don't care much about the output size of a single task, but rather about the output size of the entire stage. Allowing to drop this information at the task level allows fault tolerant execution implementation to incrementally merge these statistics and only keep them in memory at the stage level.

Reply:
Changed to `retrieveAndDrop`